### PR TITLE
Typed API extensions

### PR DESCRIPTION
**Pull request by dcoutts**
Extensions:
 * round trip tests for envelope serialisation
 * operational certificates
 * `verificationKeyHash` to get the key id
 * serialisation for key hashes
 * serialisation for addresses

### CODE REVIEW

File: cardano-api/src/Cardano/Api/Typed.hs [hunk 1]
```diff
@@ -402,37 +429,129 @@ data StakeCredential
...
+    serialiseToRawBytes (ByronAddress addr) =
+        Shelley.serialiseAddr (Shelley.AddrBootstrap addr)
 
-type StakeCredentialPointer = Shelley.Ptr
+    serialiseToRawBytes (ShelleyAddress nw pc scr) =
+        Shelley.serialiseAddr (Shelley.Addr nw pc scr)
+
+    deserialiseFromRawBytes AsShelleyAddress bs =
+        case Shelley.deserialiseAddr bs of
+          Nothing -> Nothing
```
Comment (Jimbo4350) [inline cardano-api/src/Cardano/Api/Typed.hs:473]: Why not expose the error as an `Either` rather than a `Maybe`?

Comment (erikd) [reply cardano-api/src/Cardano/Api/Typed.hs:473]: :100:

**Comment by Jimbo4350** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**